### DISCUSSION TITLE
Re-export the focus management utilities in index.ts

### DISCUSSION THREAD

**Discussion by LoganDark** [tailwindlabs/headlessui#782] [Ideas]
I think it's a bit unfair that these very useful utilities are buried under a long import, especially when it requires [configuration overrides](https://github.com/facebook/create-react-app/issues/11359) to use. If someone wants to make a multi-select box, or a tree view, or something else that Headless UI currently doesn't support yet, having the focus management utilities available would be a big help and would make it a bit more difficult to accidentally reinvent the wheel.

Alternatively, you could move this focus management stuff into another package for all to use. It's really useful!

**Comment by ragulka** [comment]
It looks like the `*.esm.js` files for these utils have been completely removed from the latest version (1.5.0), so it's no longer possible to even import them using the long path.

I understand these were not public interfaces to begin with. However, the focus management hooks were very useful - for the exact reasons stated by OP. I was using it to get a list of all focusable elements in a container, to build my own custom control/component that needed to manage focus. 

In my case, I need to set the initial focus on an element when the component is mounted. It's not exactly a Dialog, so I don't want to trap the focus. Moving the focus will simply allow for better keyboard navigation. I was using the `focusIn` hook for this, but cannot, anymore. Unfortunately, `FocusTrap` is quite strict - it cannot be even toggled on/off - even though the `useFocusTrap` hook has an `enabled` param, which could be used for this.

**Reply by jamesthomsondev** [reply]
I can definitely see use (and have had to hack my own) for overriding `FocusTrap`. My most recent use case is a Dialog (modal) that has a Combobox within it. We've set the Combobox to automatically display the list of options when the input gets focused. Due to the automatic behaviour of `FocusTrap` this was causing the Combobox input to be focused, causing the list to undesirably display over the content. Had to write a custom focus to override this after the Dialog opened - a simple solution could have been to pass `false` to `initialFocus`, but that's unsupported.
